src/components/ProductCard/AddToCartContainer.jsx

```diff
@@ -0,0 +1,80 @@
+import { useState, useCallback, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import classNames from 'classnames';
+
+import { addToCart } from '../../redux/actions/addProductToCart';
+import { updateCart } from '../../redux/actions/addProductToCart';
+import helpers from '../../helpers';
+
+export default function AddToCartContainer({ product }) {
+  const dispatch = useDispatch();
+
+  const [count, setCount] = useState(1);
+
+  const cart = useSelector((state) => state.cart);
+
+  const isInCart = useMemo(
+    () => cart.some((el) => el.id === product.id),
+    [cart, product],
+  );
+
+  const handleAddToCart = useCallback(() => {
+    const productData = {
+      id: product.id,
+      price: product.price,
+      quantity: count,
+      name: product.title,
+    };
+    if (isInCart) {
+      dispatch(updateCart(productData));
+      helpers.showInfoMessage(
+        `you have successfully added to the cart ${count} items`,
+      );
+      return;
+    }
+    dispatch(addToCart(productData));
+    helpers.showSuccessMessage(`${product.title} was added to the cart`);
+  }, [product, count, dispatch, isInCart]);
+
+  const handleDecrement = useCallback(() => {
+    if (count === 1) return;
+
+    setCount(count - 1);
```

Comment:
недавно смотрели видео про оптимизацию рендера
давай сделаем так же и будем передавать функцию
```suggestion
    setCount((currentCount) => currentCount + 1);
```